massgen/mcp_tools/hooks.py

```
@@ -821,6 +822,205 @@ async def execute(
+        if round_num == 0:
+            return self.initial_timeout_seconds
+        else:
+            return self.subsequent_timeout_seconds
+
+    def reset_for_new_round(self) -> None:
+        """Reset the hook state for a new round."""
+        self._soft_timeout_fired = False
+
+    async def execute(
+        self,
+        function_name: str,
```

Ruff (0.14.10)

Unused method argument: `function_name`

(ARG002)